### PR TITLE
Fix/fix phone regex

### LINKED ISSUE #128: Fix phone validation regex
Fix the phone regex for phone/whatsapp contact validation in `lib/cambiatus/accounts/contact.ex` to support brazilian numbers in the format +55 xx 9xxxxxxxx

### PR DESCRIPTION
**Pull request by venomnert**
## What issue does this PR close

Closes #128 

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Introduced a standard way to verify international phone numbers
- [x] Updated telegram and instagram username regex
- [x] Included test coverage for contacts

## How to test ( a list of instructions on how to test this PR)

- Run `mix test`. Ensure all test pass
- Create a contact info of `phone` type and update it `whatsapp`.
- Try entering invalid telegram and instagram ids.

Ensure to put focus on user's contact creation.

### CODE REVIEW

File: lib/cambiatus/accounts/contact.ex [hunk 1]
```diff
@@ -26,24 +33,31 @@ defmodule Cambiatus.Accounts.Contact do
...
+        validate_format(changeset, :external_id, ~r/(?s).*/)
+    end
+  end
 
-        _ ->
-          ~r/(?s).*/
-      end
+  defp validate_phone_number(changeset, country_code \\ @default_country_code) do
+    external_id = changeset.changes.external_id
+    {:ok, phone_number} = ExPhoneNumber.parse(external_id, country_code)
```
Comment (lucca65) [inline lib/cambiatus/accounts/contact.ex:56]: this way we are always checking the number as a US number, maybe we can try to pattern match / regex match the prefix to get the number? maybe expect always a `(\+){1}\d{2}` for the prefix

File: lib/cambiatus/accounts/contact.ex [hunk 2]
```diff
@@ -26,24 +33,31 @@ defmodule Cambiatus.Accounts.Contact do
...
 
-    validate_format(changeset, :external_id, regex)
+    case ExPhoneNumber.is_possible_number?(phone_number) do
```
Comment (lucca65) [inline lib/cambiatus/accounts/contact.ex:58]: I really think cases of true/false should be `if`s

**Review by lucca65 — CHANGES_REQUESTED** [review]

File: lib/cambiatus/accounts/contact.ex (hunk 1 above)
Comment (NeoVier) [reply lib/cambiatus/accounts/contact.ex:56]: shouldn't the prefix be `\+?\d{1,3}`? There are country codes that are 1 digit long (US), 2 digits long (BR), and 3 digits long (CR)

Comment (lucca65) [reply lib/cambiatus/accounts/contact.ex:56]: +1 but the `+` shouldn't be optional

Comment (venomnert) [reply lib/cambiatus/accounts/contact.ex:56]: I believe the library can automatically detect phone number and its country. In that case we probably won't need regex.

I need to dig more into this.

File: lib/cambiatus/accounts/contact.ex (hunk 2 above)
Comment (venomnert) [reply lib/cambiatus/accounts/contact.ex]: I have made the update @lucca65

File: lib/cambiatus/accounts/contact.ex (hunk 1 above)
Comment (venomnert) [reply lib/cambiatus/accounts/contact.ex:56]: Even though I set the default country code `BR` the library parses and recognizes all valid countries. The current implementation recognizes CR, US and BR and other countries, so no change is required.

File: lib/cambiatus/accounts/contact.ex [hunk 3]
```diff
@@ -26,24 +33,32 @@ defmodule Cambiatus.Accounts.Contact do
   end
 
   def validate_external_id(changeset) do
-    regex =
-      case changeset.params["type"] do
-        "phone" ->
-          ~r/^\+?\(?[0-9]{3}\)?[-\s\.]?\(?[0-9]{2}?\)?[-\s\.]?[0-9]{3,5}[-\s\.]?[0-9]{3,4}$/
+    case changeset.changes.type do
```
Comment (lucca65) [inline lib/cambiatus/accounts/contact.ex:36]: uh I didn't knew about this! nice!

File: lib/cambiatus/accounts/contact.ex (hunk 1 above)
Comment (lucca65) [reply lib/cambiatus/accounts/contact.ex:56]: oh I see! nice!

**Review by lucca65 — APPROVED** [review]

**Comment by venomnert** [comment]
- Resolved regex for social media users
- Standardized international phone number validation

**Pull request merged**